phoenix-core/src/it/java/org/apache/phoenix/end2end/CDCMiscIT.java

```diff
@@ -446,4 +846,91 @@ public void testSelectUncoveredIndex() throws Exception {
+        assertEquals(2, rs.getInt(2));
+        HashMap<String, Object> row2Json = gson.fromJson(rs.getString(3), HashMap.class);
+        String row2BinaryColStr = (String) ((Map)((Map)row2Json.get(CDC_CHANGE_IMAGE))).get("A_BINARY");
+        byte[] row2BinaryCol = Base64.getDecoder().decode(row2BinaryColStr);
+
+        assertEquals(0, DescVarLengthFastByteComparisons.compareTo(byteColumnValues.get(1),
+                0, byteColumnValues.get(1).length, row2BinaryCol, 0, row2BinaryCol.length));
+    }
+
+    @Test
+    public void testCDCBinaryAndDateColumn() throws Exception {
+        Properties props = new Properties();
+        props.put(QueryServices.TASK_HANDLING_INTERVAL_MS_ATTRIB, Long.toString(Long.MAX_VALUE));
+        props.put("hbase.client.scanner.timeout.period", "6000000");
+        props.put("phoenix.query.timeoutMs", "6000000");
+        props.put("zookeeper.session.timeout", "6000000");
+        props.put("hbase.rpc.timeout", "6000000");
+        Connection conn = DriverManager.getConnection(getUrl(), props);
+        String tableName = generateUniqueName();
+        List<byte []> byteColumnValues = new ArrayList<>();
+        byteColumnValues.add( new byte[] {0,0,0,0,0,0,0,0,0,1});
+        byteColumnValues.add(new byte[] {0,0,0,0,0,0,0,0,0,2});
+        List<Date> dateColumnValues = new ArrayList<>();
+        dateColumnValues.add(Date.valueOf("2024-02-01"));
+        dateColumnValues.add(Date.valueOf("2024-01-31"));
+        Timestamp timestampColumnValue = Timestamp.valueOf("2024-01-31 12:12:14");
+        try {
+
+            conn.createStatement().execute("CREATE TABLE  " + tableName +
+                    " ( k INTEGER PRIMARY KEY," + " a_binary binary(10), d Date, t TIMESTAMP)");
+
+            String upsertQuery = "UPSERT INTO " + tableName + " (k, a_binary, d, t) VALUES (?, ?, ?, ?)";
+            PreparedStatement stmt = conn.prepareStatement(upsertQuery);
+            stmt.setInt(1, 1);
+            stmt.setBytes(2, byteColumnValues.get(0));
+            stmt.setDate(3, dateColumnValues.get(0));
+            stmt.setTimestamp(4, timestampColumnValue);
+            stmt.execute();
+            stmt.setInt(1, 2);
+            stmt.setBytes(2, byteColumnValues.get(1));
+            stmt.setDate(3, dateColumnValues.get(1));
+            stmt.setTimestamp(4, timestampColumnValue);
+            stmt.execute();
+            conn.commit();
+
+            String cdcName = generateUniqueName();
+            String cdc_sql = "CREATE CDC " + cdcName
+                    + " ON " + tableName;
+            createAndWait(conn, tableName, cdcName, cdc_sql);
+            assertCDCState(conn, cdcName, null, 3);
+            assertCDCBinaryAndDateColumn(conn.createStatement().executeQuery
+                    ("SELECT /*+ CDC_INCLUDE(PRE, POST, CHANGE) */ * " + "FROM " + cdcName),
+                    byteColumnValues, dateColumnValues, timestampColumnValue);
+        } finally {
+            conn.close();
+        }
+    }
```

Comment:
We need to have tests with successful index writes but failed data table writes. Please see GlobalIndexCheckerIT for how to fail data table writes. We should not see any CDC event for these failed transactions.